src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1004,14 +1005,120 @@ public IEnumerable<Index> GetIndexes()
 
         public void RunIdleOperations()
         {
-            AsyncHelpers.RunSync(HandleUnusedAutoIndexes);
+            long etag;
+            DatabaseRecord record;
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+                record = _serverStore.Cluster.ReadDatabase(context, _documentDatabase.Name, out etag);
+
+            if (record.Topology.Members[0] != _serverStore.NodeTag)
+                return;
+
+            AsyncHelpers.RunSync(() => RunIdleOperationsAsync(etag));
         }
 
-        private async Task HandleUnusedAutoIndexes()
+        private async Task RunIdleOperationsAsync(long databaseRecordEtag)
         {
-            if (_serverStore.IsLeader() == false)
+            await HandleUnusedAutoIndexes();
+            await DeleteOrMergeSurpassedAutoIndexes(databaseRecordEtag);
+        }
+
+        private async Task DeleteOrMergeSurpassedAutoIndexes(long databaseRecordEtag)
+        {
+            if (_lastSurpassedAutoIndexesDatabaseRecordEtag >= databaseRecordEtag)
                 return;
 
+            _lastSurpassedAutoIndexesDatabaseRecordEtag = databaseRecordEtag;
```

Comment:
If there have been errors, we won't run until the next DatabaseRecord update.
What about moving this to the end?